client/js/statemanaged.js

```diff
@@ -36,7 +36,7 @@ export class StateManaged {
     for(const [ id, properties ] of Object.entries(this.inheritFrom()))
       if(this.inheritFromIsValid(properties, key) && widgets.has(id) && widgets.get(id).get(key) !== undefined)
         return widgets.get(id).get(key);
-    return this.defaults[key];
+    return this.defaults[key] === undefined ? null : this.defaults[key];
```

Comment:
get(property) already returns `null` if the value returned by getDefaultValue(property) is `undefined`. I'm not sure where else getDefaultValue() is used.

Reply:
The problem is here: https://github.com/ArnoldSmith86/virtualtabletop/blob/1af6e3368a8110d58c88c0027802ebe9947b6e9f/client/js/statemanaged.js#L17-L20

But I guess we either remove the logic from `get` or add it to `applyDelta` as well.

Reply:
Yes, I think moving the logic from get to getDefaultValue makes the most sense.

Reply:
Or we change `deltaForDOM[i] = this.getDefaultValue(i);` to `deltaForDOM[i] = this.get(i);`. Then it definitely has the value everyone else will see and since it was just removed from `this.state`, it should be the default anyway.

Reply:
Yes, just checking `this.get()` also makes sense.

I understand that `this.getDefaultValue()` should be a separate method so that inheriting classes can make changes to the method without needing to reimplement other parts of `this.get()` but perhaps `this.get()` should be the only place that `this.getDefaultValue()` is called.

I don't see how moving the logic from `this.get()` to `this.getDefaultValue()` could cause any isues for the JSON editor that would not also be caused by duplicating the logic into `this.getDefaultValue()`. The return value of `this.get()` would not change either way.

Reply:
> I don't see how moving the logic from `this.get()` to `this.getDefaultValue()` could cause any isues for the JSON editor that would not also be caused by duplicating the logic into `this.getDefaultValue()`. The return value of `this.get()` would not change either way.

Yes. But if we use `get` inside `applyDelta` instead, we don't have to change `get` and `getDefaultValue`. The JSON editor uses `getDefaultValue` for the command buttons. It probably wouldn't be an issue either way though.